esphome/components/tsl2591/sensor.py

```diff
@@ -0,0 +1,155 @@
+# Credit where due....
+# I put a certain amount of work into this, but a lot of ESPHome integration is
+# "look for other examples and see what they do" programming-by-example. Here are
+# things that helped me along with this:
+#
+# - I mined the existing tsl2561 integration for basic structural framing for both
+#   the code and documentation.
+#
+# - I looked at the existing bme280 integration as an example of a single device
+#   with multiple sensors.
+#
+# - Comments and code in this thread got me going with the Adafruit TSL2591 library
+#   and prompted my desired to have tsl2591 as a standard component instead of a
+#   custom/external component.
+#
+# - And, of course, the handy and available Adafruit TSL2591 library was very
+#   helpful in understanding what the device is actually talking about.
+#
+# Here is the project that started me down the TSL2591 device trail in the first
+# place: https://hackaday.io/project/176690-the-water-watcher
+
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import i2c, sensor
+from esphome.const import (
+    CONF_GAIN,
+    CONF_ID,
+    CONF_INTEGRATION_TIME,
+    CONF_FULL_SPECTRUM,
+    CONF_INFRARED,
+    CONF_VISIBLE,
+    CONF_CALCULATED_LUX,
+    DEVICE_CLASS_ILLUMINANCE,
+    STATE_CLASS_MEASUREMENT,
+    ICON_BRIGHTNESS_6,
+    UNIT_LUX,
+)
+
+# The Adafruit sensors library requires both i2c and spi in ESPhome configs, but
+# we don't use spi for this component.
+DEPENDENCIES = ["i2c", "spi"]
+
+tsl2591_ns = cg.esphome_ns.namespace("tsl2591")
+
+# This enum is a clone of the enum in the Adafruit library.
+# I couldn't work out how to use the Adafruit enum directly in codegen.
+TSL2591IntegrationTime = tsl2591_ns.enum("TSL2591IntegrationTime")
```

Comment:
```suggestion
TSL2591IntegrationTime = cg.global_ns.enum("tsl2591IntegrationTime_t")
```

This should allow you to get rid of the duplication.

Reply:
moot